File: submissions/Harini S/DAY 34/day34_queue.py
class Queue:
    def __init__(self,size):
        self.size=size
        self.queue=[None]*size
        self.front=-1
        self.rear=-1
    def enqueue(self,value):
        if self.rear<self.size-1:
            if self.front==-1:
                self.front=0
            self.rear+=1
            self.queue[self.rear]=value
        else:
            print("Queue overflow")
    
    def dequeue(self):
        if self.front==-1 or self.front>self.rear:
            print("Queue underflow")
        else:
            self.queue[self.front]=None
            self.front+=1
    
    def display(self):
        if self.front==-1 or self.front>self.rear: print("Queue underflow")
        else:
            for i in self.queue[self.front:self.rear+1]: print(i,end=" ")
lis=list(map(int,input("Input: ").split()))
deq=int(input("Dequeue: "))
q=Queue(len(lis))
for i in lis: q.enqueue(i)
for i in range(deq): q.dequeue()
print("Remaining queue:",end=" ")
q.display()

File: submissions/MohanKumarS/phase2/day34_queue.c
#include <stdio.h>
#define SIZE 20 // define the size for the Queue

int queue[SIZE];
int front = -1, rear = -1; // define the front and rear points

// Enqueue
void enqueue(int value)
{
    if (rear == SIZE - 1)
    {
        printf("Queue Overflow\n");
        return;
    }
    if (front == -1) 
        front = 0;
    rear = rear + 1;
    queue[rear] = value;
}

// Dequeue 
void dequeue()
{
    if (front == -1 || front > rear) 
    {
        printf("Queue Underflow\n");
        return;
    }
    front = front + 1;
}

// Display function
void display()
{
    if (front == -1 || front > rear) 
    {
        printf("Queue is empty.\n");
        return;
    }
    printf("Current Queue: ");
    for (int i = front; i <= rear; i++) 
    {
        printf("%d ", queue[i]);
    }
    printf("\n");
}

int main() {
    enqueue(10);
    enqueue(20);
    enqueue(30);
    enqueue(40);
    display();
    dequeue();
    dequeue();
    display();

    return 0;
}
